### PR TITLE
Feature  sort

### PR DESCRIPTION
**Pull request by bosulliv42**

### CODE REVIEW

File: controllers/api/v1/datas.controller.js [hunk 1]
```diff
@@ -120,14 +122,20 @@ function ServicesController(options) {
...
           return next({code: 400, messages: query["ERRORS"]});
         var queryOptions = {
-          sort: {updatedAt: -1, createdAt: -1},
           populate: []
         };
+        if (res._request !== undefined && res._request.params !== undefined && res._request.params.sort !== undefined) {
+          queryOptions = SortService.buildMongoQuery(queryOptions, res._request.params.sort, 'Data', whitelistedFields);
+        }
+        else {
+          queryOptions["sort"] = {updatedAt: -1, createdAt: -1};
```
Comment (jgigault) [inline controllers/api/v1/datas.controller.js]: this default value could be placed in the service when `res._request.params.sort` is undefined

File: services/sort.service.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  else {
+    return (undefined);
+  }
+}
+
+function SortService(options) {
+  function checkSortParam(sorts, param)
+  {
+    var objRes = {};
+    if (typeof sorts === 'object')
```
Comment (jgigault) [inline services/sort.service.js]: sorts is not an object, it is a string

eg: "?sort=created_at"
eg: "?sort=name"
eg: "?sort=-name" (inversed order)

Comment (jgigault) [inline services/sort.service.js]: Or again:  "?sort=public,created_at,-updated_at" (multiple sorting keys)

File: controllers/api/v1/datas.controller.js (hunk 1 above)
Comment (bosulliv42) [inline controllers/api/v1/datas.controller.js]: yes but all default parameter are not the same in route field data and service

Comment (jgigault) [inline controllers/api/v1/datas.controller.js]: It should be the same ;)
All models implement twice attributes createdAt and updatedAt

Comment (bosulliv42) [inline controllers/api/v1/datas.controller.js]: ok

File: controllers/api/v1/datas.controller.js [hunk 3]
```diff
@@ -130,12 +130,9 @@ function ServicesController(options) {
         var queryOptions = {
```
Comment (jgigault) [inline controllers/api/v1/datas.controller.js]: You can remove this instantiation

File: services/sort.service.js [hunk 4]
```diff
@@ -49,24 +49,30 @@ function SortService(options) {
...
-          else {
-            param["errors"].push("the name " + key + " is not in the field model");
-          }
-        } else {
-          param["errors"].push("the value for " + key +" is not -1 or 1");
+      var tab = sorts.split(',');
+      Object.keys(tab).every(function (key) {
+        
+        var sign = 1;
+        if (tab[key][0] === '-') {
```
Comment (jgigault) [inline services/sort.service.js]: What about "+" ?
(eg.: "?filter=+name")

File: services/sort.service.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const CONFIG = require('../config/app.js');
+
+module.exports = SortService;
+
+function getTabByKeyVal(key, val, myArray, myarray2) {
```
Comment (jgigault) [inline services/sort.service.js]: why is it globalized outside the service ?

File: services/sort.service.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        var sign = 1;
+        if (tab[key][0] === '-') {
+          sign = -1;
+          tab[key] = tab[key].substring(1);
+        }
+        else if (tab[key][0] === '+') {
+          tab[key] = tab[key].substring(1);
+        }
+        if (tab[key] in objRes) {
+          param["errors"].push("you allready set a sort by the field" + tab[key] + " you can only set one type of sort for one field");
```
Comment (jgigault) [inline services/sort.service.js]: could you replace with something shorter like: 
```
"sort by field \"" + tab[key] + "\" has already been set"
```

File: services/sort.service.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        }
+        else if (getValIfValExistInArray(tab[key], param)) {
+          objRes[tab[key]] = sign;
+        }
+        else {
+          param["errors"].push("the name " + tab[key] + " is not in the field model");
```
Comment (jgigault) [inline services/sort.service.js]: please replace "the name" with "the field" and enclose it within double-quotes
you can say "is not in the model"

File: services/sort.service.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        }
+        return true;
+      });
+    }
+    else {
+      param["errors"].push("sort need to be a string : ?sort=value1 or ?sort=-value1 or ?sort=value1,value2,value3");
```
Comment (jgigault) [inline services/sort.service.js]: no space between ":" and the preceding word

File: services/sort.service.js (hunk 5 above)
Comment (bosulliv42) [inline services/sort.service.js]: my bad ;)

File: services/sort.service.js (hunk 6 above)
Comment (bosulliv42) [inline services/sort.service.js]: ok

**Pull request merged**